helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -222,13 +217,21 @@ public int getAssignedReplicaCount() {
   /**
    * Return the most concerning capacity utilization number for evenly partition assignment.
    * The method dynamically returns the highest utilization number among all the capacity
-   * categories.
+   * categories assuming the new capacity usage is added to the node.
    * For example, if the current node usage is {CPU: 0.9, MEM: 0.4, DISK: 0.6}. Then this call shall
    * return 0.9.
+   * @param newUsage the proposed new additional capacity usage.
    * @return The highest utilization number of the node among all the capacity category.
    */
-  public float getHighestCapacityUtilization() {
-    return _highestCapacityUtilization;
+  public float getExpectedHighestUtilization(Map<String, Integer> newUsage) {
```

Comment:
When I first saw this name, I was confused. I think it would be clearer if you renamed this function to:

getProjectedHighestUtil... (preferred)
or 
getEstimatedHighestUtil...

Is this mainly to reduce duplicate code?

Reply:
No, the previous one has a bug if you look into deep.
I like the name, will change it later.